page-objects/pages/TASKListPage.js

```diff
@@ -201,18 +203,26 @@ async function getNuxtTaskList() {
 
 async function hoverOverTaskAndClickMenu(taskName) {
 	await driver.pause(5000);
-	// to be refactored, we shouldn't create selectors like that, we should try to use mod_extsprintf as in TASKPage function isTaskGraded
-	await driver.$(taskTitleText + taskName + "']").moveTo();
-	await driver.pause(5000);
-	// to be refactored, we shouldn't create selectors like that, we should try to use mod_extsprintf as in TASKPage function isTaskGraded
-	await elementHelpers.click(taskActionMenu + taskName + "']");
-	await driver.pause(3000);
+	let taskTitle = await driver.$(mod_extsprintf.sprintf(taskTitleText, taskName));
+	await taskTitle.scrollIntoView(false);
+	if (await taskTitle.isDisplayedInViewport()){
+		let xOffset = await taskTitle.getLocation('x');
+		let yOffset = await taskTitle.getLocation('y');
+		taskTitle.moveTo(xOffset, yOffset);
+		await driver.pause(5000);
+		await elementHelpers.click(mod_extsprintf.sprintf(taskActionMenu, taskName));
+		await driver.pause(3000);
+	}
 }
 
 async function clickTaskEditAction() {
 	await elementHelpers.hoverOverMenuOptions(editButton);
 }
 
+async function clickTaskCompletedAction() {
```

Comment:
Instead of creating a new function for each button, we should try to create one, that can use any button wanted. See const taskButton and function getTaskActionBtnSelector in TASKListPage.js.